pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go

```diff
@@ -320,6 +320,14 @@ func (s *SQLTranslator) generateSpanConfigurationsForTable(
 			"expected table descriptor, but got descriptor of type %s", desc.DescriptorType(),
 		)
 	}
+
+	// Views aren't physical tables and the range corresponding to their
+	// descriptor ID doesn't contain any data. As such, we don't create a span
+	// config record (and in-turn a split point) for them.
+	if desc.(catalog.TableDescriptor).IsView() {
```

Comment:
rather than all of this type assertion, how about you just make a variable `table := desc.(catalog.TableDescriptor`?

`IsView` is wrong because it would include materialized views, I do believe. I think you want `IsPhysicalTable`. Can you add a test that materialized views do get split off?